### DISCUSSION TITLE
Walkforward?

### DISCUSSION THREAD

**Discussion by eXfake** [kernc/backtesting.py#497] [Q&A]
Hi,
What do you recommend for walkforward?
I tried slice the data and push it to backtest. But I realized that I need more data than sliced data. 
Because to warm-up the indicators. In both sides. in-sample, out-of-sample.
Then I thought of pushing all the data and adding a trade date filter. But I couldn't figure out how to send the walkforward date loop to the strategy.
Or any idea?
Sorry for my english.
Thanks.

**Comment by kernc** [comment]
Must admit, without code, I don't quite understand what you mean. Have you seen the official [*backtesting.py tutorials*](https://kernc.github.io/backtesting.py/doc/backtesting/index.html#tutorials)? One of them shows an example of walk-forward optimization ...

**Reply by ddm-j** [reply]
@kernc  - I believe you are referring to your **Trading with ML** example? In this example, I don't think there are indicators that have different warmup periods for each separate optimization.

For instance, say you have a single indicator 200 EMA strategy. If you want to have a walk forward optimization with 200 days training, 100 days trading, you would have exactly zero trades to optimize over for each training period because the 200 EMA will not be "warmed up" until the testing period starts. Additionally, you would have zero trades in the test period.

OP is wondering what the recommendation would be, since for each optimization (say you are testing EMA 70-150 in increments of 10) you would need to pad the front of both the train and test sets with an adequate warmup period according to that iterations indicator values.

I'm new around here, so maybe there is a better way to think about this.
